pkg/finder/podresourcefinder.go

```diff
@@ -0,0 +1,480 @@
+package finder
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	// "github.com/davecgh/go-spew/spew"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	// "k8s.io/kubernetes/pkg/kubelet/util"
+
+	"github.com/fromanirh/numalign/pkg/topologyinfo/cpus"
+	"github.com/fromanirh/numalign/pkg/topologyinfo/pcidev"
+	podresources "k8s.io/kubernetes/pkg/kubelet/apis/podresources"
+	podresourcesapi "k8s.io/kubernetes/pkg/kubelet/apis/podresources/v1alpha1"
+	v1alpha1 "github.com/swatisehgal/topologyapi/pkg/apis/topology/v1alpha1"
+)
+
+const (
+	defaultPodResourcesTimeout = 10 * time.Second
+	defaultPodResourcesMaxSize = 1024 * 1024 * 16 // 16 Mb
+)
+
+// type CGroupPathTranslator func(sysfs, cgroupPath string) string
+
+type PodResourceFinder struct {
+	args            Args
+	// conn            *grpc.ClientConn
+	// client          pb.RuntimeServiceClient
+	podResourceClient podresourcesapi.PodResourcesListerClient
+	// cgroupTranslate CGroupPathTranslator
+	// we may want to move to cadvisor past PoC stage
+	pciDevs         *pcidev.PCIDevices
+	cpus            *cpus.CPUs
+	cpuID2NUMAID    map[int]int
+	pciAddr2NUMAID  map[string]int
+	perNUMACapacity map[int]map[v1.ResourceName]int64
+	// pciaddr -> resourcename
+	pci2ResourceMap map[string]string
+}
+
+// type ContainerInfo struct {
+// 	sandboxID      string              `json:"sandboxID"`
+// 	Pid            uint32              `json:"pid"`
+// 	Removing       bool                `json:"removing"`
+// 	SnapshotKey    string              `json:"snapshotKey"`
+// 	Snapshotter    string              `json:"snapshotter"`
+// 	RuntimeType    string              `json:"runtimeType"`
+// 	RuntimeOptions interface{}         `json:"runtimeOptions"`
+// 	Config         *pb.ContainerConfig `json:"config"`
+// 	RuntimeSpec    *runtimespec.Spec   `json:"runtimeSpec"`
+// }
+
+
+func NewPodResourceFinder(args Args, pciResMapConf map[string]string) (Finder, error) {
+	finderInstance := &PodResourceFinder{
```

Comment:
if we need all this (up until line 114), can we factor this out and avoid duplication?

Reply:
+1. I have moved this to finder.go. Let me know if looks better now